plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -3844,68 +3844,27 @@ public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session,
-        if (table.isEmpty() || isSomeKindOfAView(table.get())) {
-            return Optional.empty();
-        }
-
-        Optional<CatalogSchemaTableName> catalogSchemaTableName = Optional.<CatalogSchemaTableName>empty()
-                .or(() -> redirectTableToIceberg(icebergCatalogName, table.get()))
-                .or(() -> redirectTableToDeltaLake(deltaLakeCatalogName, table.get()))
-                .or(() -> redirectTableToHudi(hudiCatalogName, table.get()));
-
-        // stitch back the suffix we cut off.
-        return catalogSchemaTableName.map(name -> new CatalogSchemaTableName(
-                name.getCatalogName(),
-                new SchemaTableName(
-                        name.getSchemaTableName().getSchemaName(),
-                        name.getSchemaTableName().getTableName() + tableNameSplit.getSuffix().orElse(""))));
-    }
+        SchemaTableName schemaTableName = new SchemaTableName(tableName.getSchemaName(), tableNameSplit.getBaseTableName());
+        Optional<CatalogSchemaTableName> catalogSchemaTableName = tableRedirectionsProvider.redirectTable(session, schemaTableName, table);
 
-    private Optional<CatalogSchemaTableName> redirectTableToIceberg(Optional<String> targetCatalogName, Table table)
-    {
-        if (targetCatalogName.isEmpty()) {
-            return Optional.empty();
-        }
-        if (isIcebergTable(table)) {
-            return targetCatalogName.map(catalog -> new CatalogSchemaTableName(catalog, table.getSchemaTableName()));
+        if (catalogSchemaTableName.isPresent()) {
+            String suffix = tableNameSplit.getSuffix().orElse("");
+            return catalogSchemaTableName.map(name -> appendSuffixToTableName(name, suffix));
         }
-        return Optional.empty();
-    }
 
-    private Optional<CatalogSchemaTableName> redirectTableToDeltaLake(Optional<String> targetCatalogName, Table table)
-    {
-        if (targetCatalogName.isEmpty()) {
+        if (table.isEmpty() || isSomeKindOfAView(table.get())) {
             return Optional.empty();
         }
-        if (isDeltaLakeTable(table)) {
-            return targetCatalogName.map(catalog -> new CatalogSchemaTableName(catalog, table.getSchemaTableName()));
-        }
-        return Optional.empty();
-    }
 
-    private Optional<CatalogSchemaTableName> redirectTableToHudi(Optional<String> targetCatalogName, Table table)
-    {
-        if (targetCatalogName.isEmpty()) {
-            return Optional.empty();
-        }
-        if (isHudiTable(table)) {
-            return targetCatalogName.map(catalog -> new CatalogSchemaTableName(catalog, table.getSchemaTableName()));
-        }
-        return Optional.empty();
+        // stitch back the suffix we cut off.
+        String suffix = tableNameSplit.getSuffix().orElse("");
+        return catalogSchemaTableName.map(name -> appendSuffixToTableName(name, suffix));
```

Comment:
I was reviewing this to see if we could simplify it at all and realized that it is actually identical to:
```suggestion
        Optional<CatalogSchemaTableName> catalogSchemaTableName = tableRedirectionsProvider.redirectTable(session, schemaTableName, table);

        // stitch back the suffix we cut off.
        String suffix = tableNameSplit.getSuffix().orElse("");
        return catalogSchemaTableName.map(name -> appendSuffixToTableName(name, suffix));
```
Basically if `redirectTable` returns non-empty, we return it, else we always return `Optional.empty()` (due to the new `if (catalogSchemaTableName.isPresent())` clause). So the intermediate stuff is irrelevant.

I think we _do_ still want the `isSomeKindOfAView(table.get())` logic, but in `DefaultHiveTableRedirectionsProvider`, instead of here, like:
```java
    public Optional<CatalogSchemaTableName> redirectTable(ConnectorSession session, SchemaTableName name, Optional<Table> table)
    {
        return table.filter(t -> !isSomeKindOfAView(t)).flatMap(t -> {
          ...
        });
    }
```

Reply:
Yes, we should do it in the suggested way. Made the change.